flang/lib/Lower/Allocatable.cpp

```diff
@@ -55,7 +55,7 @@ struct ErrorManager {
             const Fortran::lower::SomeExpr *statExpr,
             const Fortran::lower::SomeExpr *errMsgExpr) {
     Fortran::lower::StatementContext stmtCtx;
-    auto &builder = converter.getFirOpBuilder();
+    fir::FirOpBuilder &builder = converter.getFirOpBuilder();
```

Comment:
Straddles the fuzzy line a bit. The `getFoo` is sometimes enough, sometimes not, depending on the reviewer.

Reply:
Does Mehdi accept "auto" here?

Reply:
Do you know whether Mehdi would reject `auto` when the initializer is a call to `get<TYPE>()`?

Reply:
They accept `auto` in MLIR where they have `auto loc = op.getLoc()` and the actual type should be `mlir::Location`. 

`auto` should be fine when the initializer is a call to `get<TYPE>()`.